### ISSUE TITLE
solidify and document JSON schemas

### ISSUE THREAD

**Issue by willkg** [pyvideo/pyvideo-data#21]
We have two kinds of things we're tracking now: conferences (category.json) and videos (_slug_.json).

There are tentative schemas for both of these that are tied into the validation code here:

https://github.com/pyvideo/pyvideo-data/blob/8b8e19e70300c25e742a931d9190f76d10daca17/src/clive/validate.py#L155

This issue covers:
1. discussing those schemas on the mailing list and establishing a version 1 of those schemas
2. figuring out whether we want to do schema versioning at this time
3. figuring out how to document the schemas because clearly using the code as documentation is pretty meh

**Comment by logston** [comment]
1. \* I feel like the schema provided by the link above is a great start. 
2. If we do use schema versioning, then we don't have to stress so much about setting down the perfect one now and we can just roll with what we have at the moment. I'm +1 for versioning.
3. Has http://json-schema.org/ or the like been considered? I know this would not handle the validation of the content specified at certain keys (like description which requires markup validation, see #22). But it might be a nice way to offload the work of validating the JSON object structure and allow us to focus on validation of the content. Full disclosure, I've never worked with a JSON schema before but I am interested. 
- Should I be discussing this via the mailing list? Can we link people to this issue who raise the topic in the mailing list?

**Comment by willkg** [comment]
To your points quickly:
1. Yay!
2. I'll add schema versioning. I think I need to tweak how validation works first so that we can maintain multiple schema versions. That's covered in issue #26.
3. I didn't think about using json-schema. I skimmed the site and it doesn't seem obvious that it really helps us. Amongst other things, we have a date primitive that it's lacking. I'd like to stick with my schema and validator. If someone else wants to write up a new issue for switching to json-schema and do the leg work for looking into doing everything we're doing using json-schema, I'd be interested in seeing the outcome of that.

Issues should be discussed in the issue tracker. I'm pretty sure we have two different sets of people following issues vs. the mailing list. If you want to bring it up on the mailing list, that's cool. With other things, @codersquid has done that. I don't think anyone has pointed out this issue, yet.

I think it'd be ok to get this version of the schema done with the expectation that as more people start using it, other requirements will surface and we'll have more people involved in future iterations of the schema.

**Comment by logston** [comment]
Okay. Sounds good. Thanks for the response.

**Comment by willkg** [comment]
I spent more time looking at json-schema. Seems like switching to it is a good idea, but we'd need to figure out how to do some of the data validation we need to do. Need to figure out if there's a way to encode properties or something that denotes additional data validation for fields.

Anyhow, if we switched to json-schema, the following libs would be useful:
- https://github.com/Julian/jsonschema : for validating documents against a schema
- http://jsl.readthedocs.io/en/latest/index.html : for describing schemas using Python (this might be nice, but it's not 100% necessary)

I'll try to get further with this soon.

**Comment by willkg** [comment]
I spent some time with json schema and then decided I didn't feel like dealing with it (again).

So I went ahead and fixed the issues with the current schema system. We may still outgrow it. That's fine. We can deal with that bridge when we get there.

I added schema documentation in #133. That's autogenerated from the schema definition. It's not done, but I think there's enough there to show the direction I'm tentatively heading in.

Any help on that PR would be really super duper.
